### ISSUE TITLE
Can't compile for iOS

### ISSUE THREAD

**Issue by c0c4i** [CodeMinion/another_brother#8]
Hello, I'm trying to running my app that use `another_brother` on iOS but I have some troubles.
I have add lines to `info.plist` file but still doesn't work.
I have also followed youtube video for `libBROTHERSDK.a`

This is the error
```
2 warnings generated.
3
    Command CompileSwift failed with a nonzero exit code
    Undefined symbols for architecture x86_64:
      "_OBJC_CLASS_$_BROTHERSDK", referenced from:
          objc-class-ref in TbStartCommunicationMethodCall.o
    ld: symbol(s) not found for architecture x86_64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete
    warning: no rule to process file '.../ios/Pods/BROTHERSDK/BROTHERSDK/libBROTHERSDK.a' of type 'archive.ar' for architecture 'x86_64' (in target 'BROTHERSDK' from project 'Pods')
```

**Comment by CodeMinion** [comment]
Hello @c0c4i ,

Sorry to see you are having build issues and thanks for the logs. I see it mentions x86_64 so it feels you might be building with the simulator as a target. Could you confirm if that is the case? 

If so can you try building with a physical iOS device and see if you still experience any build issues? 

Thanks in advance,

**Comment by c0c4i** [comment]
Hello, yes I was trying to build for simulator.
Today I tried with physical iOS device but I have the same issue.
Also I can't archive app to release on store.

**Comment by CodeMinion** [comment]
Thanks for confirming @c0c4i. I would suggest taking a look again at the check from the video because when you change build targets xCode may reset it at times so while it may be failing to build the error might be a different one. 

Video: https://www.youtube.com/watch?v=42dBVdXU8gk

As for the release issue it's likely linked to the errors you are seeing during build so it might let you after you get a build.

If you still get an error after building to an iPhone could you share the error here to have a look? 

**Note:** I recently also checked it building on an M1 so both the regular Macs and M1 should build fine but if you can please include your OS version along with the error.  https://www.youtube.com/watch?v=kCLR7ACkMOw

Thanks in advance,

**Comment by c0c4i** [comment]
I followed youtube video but stil doesn't work, I created a new flutter project to test it and these are the errors.
I hope you can help me, thanks.
```
Running Xcode build...                                                  
 └─Compiling, linking and signing...                        998ms
Xcode build done.                                           19,9s
Failed to build iOS app
Error output from Xcode build:
↳
    2021-06-29 09:41:06.537 xcodebuild[59975:560058] CFURLRequestSetHTTPCookieStorageAcceptPolicy_block_invoke: no longer implemented and
    should not be called
    ** BUILD FAILED **


Xcode's output:
↳
    While building module 'another_brother' imported from
    /Users/samuelebesoli/Documents/test/printer/ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/another_brothe
    r-umbrella.h:14:
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/BrotherManager
    .h:12:9: error: include of non-modular header inside framework module 'another_brother.BrotherManager':
    '/Users/samuelebesoli/Documents/test/printer/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRLMPrinterKit.h
    ' [-Werror,-Wnon-modular-include-in-framework-module]
    #import <BRLMPrinterKit/BRLMPrinterKit.h>
            ^
    While building module 'another_brother' imported from
    /Users/samuelebesoli/Documents/test/printer/ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/another_brothe
    r-umbrella.h:15:
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/BrotherUtils.h
    :12:9: error: include of non-modular header inside framework module 'another_brother.BrotherUtils':
    '/Users/samuelebesoli/Documents/test/printer/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRLMPrinterKit.h
    ' [-Werror,-Wnon-modular-include-in-framework-module]
    #import <BRLMPrinterKit/BRLMPrinterKit.h>
            ^
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/BrotherUtils.h
    :13:9: error: include of non-modular header inside framework module 'another_brother.BrotherUtils':
    '/Users/samuelebesoli/Documents/test/printer/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRLMPrinterDefin
    e.h' [-Werror,-Wnon-modular-include-in-framework-module]
    #import <BRLMPrinterKit/BRLMPrinterDefine.h>
            ^
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/BrotherUtils.h
    :14:9: error: include of non-modular header inside framework module 'another_brother.BrotherUtils':
    '/Users/samuelebesoli/Documents/test/printer/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrinterK
    it.h' [-Werror,-Wnon-modular-include-in-framework-module]
    #import <BRLMPrinterKit/BRPtouchPrinterKit.h>
            ^
    While building module 'another_brother' imported from
    /Users/samuelebesoli/Documents/test/printer/ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/another_brothe
    r-umbrella.h:18:
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/GetBluetoothPr
    intersMethodCall.h:12:9: error: include of non-modular header inside framework module
    'another_brother.GetBluetoothPrintersMethodCall':
    '/Users/samuelebesoli/Documents/test/printer/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrinterK
    it.h' [-Werror,-Wnon-modular-include-in-framework-module]
    #import <BRLMPrinterKit/BRPtouchPrinterKit.h>
            ^
    While building module 'another_brother' imported from
    /Users/samuelebesoli/Documents/test/printer/ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/another_brothe
    r-umbrella.h:19:
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/GetNetPrinters
    MethodCall.h:12:9: error: include of non-modular header inside framework module 'another_brother.GetNetPrintersMethodCall':
    '/Users/samuelebesoli/Documents/test/printer/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrinterK
    it.h' [-Werror,-Wnon-modular-include-in-framework-module]
    #import <BRLMPrinterKit/BRPtouchPrinterKit.h>
            ^
    While building module 'another_brother' imported from
    /Users/samuelebesoli/Documents/test/printer/ios/Runner/GeneratedPluginRegistrant.m:8:
    In file included from <module-includes>:1:
    In file included from
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/another_brothe
    r-umbrella.h:26:
    In file included from
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/TbBarcodeMetho
    dCall.h:12:
    /Users/samuelebesoli/Documents/test/printer/build/ios/Debug-iphoneos/another_brother/another_brother.framework/Headers/TbPrinterInfo.
    h:11:9: error: include of non-modular header inside framework module 'another_brother.TbPrinterInfo':
    '/Users/samuelebesoli/Documents/test/printer/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRLMPrinterKit.h
    ' [-Werror,-Wnon-modular-include-in-framework-module]
    #import <BRLMPrinterKit/BRLMPrinterKit.h>
            ^
    7 errors generated.
    /Users/samuelebesoli/Documents/test/printer/ios/Runner/GeneratedPluginRegistrant.m:8:9: fatal error: could not build module
    'another_brother'
    #import <another_brother/AnotherBrotherPlugin.h>
     ~~~~~~~^
    8 errors generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete
    warning: no rule to process file '/Users/samuelebesoli/Documents/test/printer/ios/Pods/BROTHERSDK/BROTHERSDK/libBROTHERSDK.a' of type
    'archive.ar' for architecture 'arm64' (in target 'BROTHERSDK' from project 'Pods')
```

**Comment by c0c4i** [comment]
I tried to archive for app store and this was the error.
```
 bitcode bundle could not be generated because '/Users/samuelebesoli/Documents/test/printer/ios/Pods/BROTHERSDK/BROTHERSDK/libBROTHERSDK.a(BROTHERSDK.o)' was built without full bitcode. All object files and libraries for bitcode must be generated from Xcode Archive or Install build for architecture armv7
```

**Comment by CodeMinion** [comment]
Thanks for the logs @c0c4i . As for the build error you are seeing it seems you might have not selected "allow non-modular includes" in your project. 

On your Runner (this is your app's runner) make sure to mark: Allow non-modular includes as yes.

Can you try again with this checked?

**Comment by c0c4i** [comment]
I solve it, thank you so much and sorry for the problem.

**Comment by CodeMinion** [comment]
No apologies necessary @c0c4i building on a Mac is always a bit of a pain at first so I always dread the times when I have to do it. 

As for the app store problem it seems to be that the libBROTHERSDK might not have all that it's required by Apple to publish to the store. If the app you are planning to does not plan to target the RJ-Go type printers which are what libBROTHERSDK handles I can releases a side branch this weekend that does not include this in iOS. 

Let me know,

**Comment by c0c4i** [comment]
Sure, you would help me a lot.
The only printer I am using is QL820 so I think there is not problem.
Thank you so much

**Comment by CodeMinion** [comment]
Hi @c0c4i , 

Just pushed a new branch (**ios-no-type-b**) which removes all RJ-Go support from the library on iOS which should allow you to publish to the app store. 

In order to use this version you will instead need to reference another_brother from GitHub rather than the pub.dev site like so:

`another_brother:
    git:
      url: git://github.com/CodeMinion/another_brother.git
      ref: ios-no-type-b
`
Let me know how it goes.

**Comment by c0c4i** [comment]
Hi, I tried it but I can't run on iPad from VS Code.
This is the error on xcode while archive.
```
.../development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/AnotherBrotherPlugin.h:2:9: 'BROTHERSDK/BROTHERSDK.h' file not found
```
This is the full error from terminal.
```
Xcode's output:
↳
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/S
    tartCommunicationMethodCall.m:9:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/S
    tartCommunicationMethodCall.h:12:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/B
    rotherUtils.h:14:
    In file included from
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrin
    terKit.h:12:
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchBLEM
    anager.h:18:3: warning: '@return' command used in a comment that is attached to a function returning void [-Wdocumentation]
     @return NO if BLE is not available, otherwise YES
     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/S
    tartCommunicationMethodCall.m:30:44: warning: unused variable 'dartPrintInfo' [-Wunused-variable]
        NSDictionary<NSString *, NSObject *> * dartPrintInfo = _call.arguments[@"printInfo"];
                                               ^
    2 warnings generated.
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/P
    rintPdfFileMethodCall.m:9:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/P
    rintPdfFileMethodCall.h:13:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/B
    rotherUtils.h:14:
    In file included from
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrin
    terKit.h:12:
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchBLEM
    anager.h:18:3: warning: '@return' command used in a comment that is attached to a function returning void [-Wdocumentation]
     @return NO if BLE is not available, otherwise YES
     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 warning generated.
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/P
    rintImageMethodCall.m:9:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/P
    rintImageMethodCall.h:12:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/B
    rotherUtils.h:14:
    In file included from
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrin
    terKit.h:12:
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchBLEM
    anager.h:18:3: warning: '@return' command used in a comment that is attached to a function returning void [-Wdocumentation]
     @return NO if BLE is not available, otherwise YES
     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 warning generated.
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/P
    rintFileMethodCall.m:9:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/P
    rintFileMethodCall.h:12:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/B
    rotherUtils.h:14:
    In file included from
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrin
    terKit.h:12:
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchBLEM
    anager.h:18:3: warning: '@return' command used in a comment that is attached to a function returning void [-Wdocumentation]
     @return NO if BLE is not available, otherwise YES
     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 warning generated.
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/P
    rintFileListMethodCall.m:9:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/P
    rintFileListMethodCall.h:13:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/B
    rotherUtils.h:14:
    In file included from
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrin
    terKit.h:12:
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchBLEM
    anager.h:18:3: warning: '@return' command used in a comment that is attached to a function returning void [-Wdocumentation]
     @return NO if BLE is not available, otherwise YES
     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 warning generated.
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/G
    etPrinterStatusMethodCall.m:9:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/G
    etPrinterStatusMethodCall.h:12:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/B
    rotherUtils.h:14:
    In file included from
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrin
    terKit.h:12:
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchBLEM
    anager.h:18:3: warning: '@return' command used in a comment that is attached to a function returning void [-Wdocumentation]
     @return NO if BLE is not available, otherwise YES
     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 warning generated.
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/G
    etNetPrintersMethodCall.m:9:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/G
    etNetPrintersMethodCall.h:12:
    In file included from
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrin
    terKit.h:12:
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchBLEM
    anager.h:18:3: warning: '@return' command used in a comment that is attached to a function returning void [-Wdocumentation]
     @return NO if BLE is not available, otherwise YES
     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 warning generated.
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/G
    etBluetoothPrintersMethodCall.m:9:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/G
    etBluetoothPrintersMethodCall.h:12:
    In file included from
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrin
    terKit.h:12:
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchBLEM
    anager.h:18:3: warning: '@return' command used in a comment that is attached to a function returning void [-Wdocumentation]
     @return NO if BLE is not available, otherwise YES
     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 warning generated.
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/E
    ndCommunicationMethodCall.m:9:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/E
    ndCommunicationMethodCall.h:12:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/B
    rotherUtils.h:14:
    In file included from
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrin
    terKit.h:12:
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchBLEM
    anager.h:18:3: warning: '@return' command used in a comment that is attached to a function returning void [-Wdocumentation]
     @return NO if BLE is not available, otherwise YES
     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/E
    ndCommunicationMethodCall.m:30:44: warning: unused variable 'dartPrintInfo' [-Wunused-variable]
        NSDictionary<NSString *, NSObject *> * dartPrintInfo = _call.arguments[@"printInfo"];
                                               ^
    2 warnings generated.
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/C
    ancelMethodCall.m:9:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/C
    ancelMethodCall.h:12:
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/Method/B
    rotherUtils.h:14:
    In file included from
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchPrin
    terKit.h:12:
    /Users/samuelebesoli/Documents/veez-app/business/ios/Pods/BRLMPrinterKit/BRLMPrinterKit/BRLMPrinterKit.framework/Headers/BRPtouchBLEM
    anager.h:18:3: warning: '@return' command used in a comment that is attached to a function returning void [-Wdocumentation]
     @return NO if BLE is not available, otherwise YES
     ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    1 warning generated.
    In file included from
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/AnotherB
    rotherPlugin.m:1:
    /Users/samuelebesoli/development/Flutter/.pub-cache/git/another_brother-bda5f212dcad1ad6ac9f8b7555f37b8f30739b92/ios/Classes/AnotherB
    rotherPlugin.h:2:9: fatal error: 'BROTHERSDK/BROTHERSDK.h' file not found
    #import <BROTHERSDK/BROTHERSDK.h>
            ^~~~~~~~~~~~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete
    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/Reachabil
    ity/Reachability.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/url_launc
    her/url_launcher.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/path_prov
    ider/path_provider.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/device_in
    fo/device_info.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/shared_pr
    eferences/shared_preferences.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/MTBBarcod
    eScanner/MTBBarcodeScanner.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/Pods_Runn
    er.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/advance_p
    df_viewer/advance_pdf_viewer.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/webview_f
    lutter/webview_flutter.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/sqflite/s
    qflite.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/FMDB/FMDB
    .framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/Runner.ap
    p'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/flutter_b
    lue/flutter_blue.framework'

    warning: Stale file
    '/Users/samuelebesoli/Library/Developer/development/Flutter/.pub-cache/hosted/pub.dartlang.org/flutter_blue-0.8.0/ios/gen/Flutterblue
    .pbobjc.h' is located outside of the allowed root paths.

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/connectiv
    ity_plus/connectivity_plus.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/qr_code_s
    canner/qr_code_scanner.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/another_b
    rother/another_brother.framework'

    note: Removed stale file
    '/Users/samuelebesoli/Library/Developer/Xcode/DerivedData/Runner-bcnbhlckpulwvpaebckepsjtybwy/Build/Products/Debug-iphoneos/Protobuf/
    protobuf.framework'
```

**Comment by c0c4i** [comment]
Hi, do you have any news?

**Comment by CodeMinion** [comment]
Hi @c0c4i ! 

Apologies for the late reply and the error. Thanks for the logs, I just pushed an update to the ios-no-type-b branch that will resolve the error. 

Since you are using the Git reference make sure to:
1. Run Flutter Clean
2. Run Pub upgrade (When it comes to the Git reference 'Pub get' will just fetch it once)

I made a full build and it all checks out but don't hesitate to let me know if you run into any issues.
